fortran/utils_ppser.f90

```diff
@@ -50,26 +50,30 @@ MODULE utils_ppser
 
 !============================================================================
 
-SUBROUTINE ppser_initialize(directory, prefix, mode, prefix_ref)
+SUBROUTINE ppser_initialize(directory, prefix, mode, prefix_ref, mpi_rank)
   CHARACTER(LEN=*), INTENT(IN)     :: directory, prefix
   INTEGER, OPTIONAL, INTENT(IN)    :: mode
   CHARACTER(LEN=*), OPTIONAL, INTENT(IN)     :: prefix_ref
   REAL                             :: realvalue
   INTEGER                          :: intvalue
+  INTEGER, OPTIONAL, INTENT(IN)    :: mpi_rank
   CHARACTER(LEN=1), DIMENSION(128) :: buffer
-  CHARACTER(LEN=6)                 :: suffix
+  CHARACTER(LEN=7)                 :: suffix
 
   ! Initialize serializer and savepoint
   IF ( .NOT. ppser_initialized ) THEN
-    ! IF (mode == 0) THEN
-      CALL fs_create_serializer(directory, TRIM(prefix), 'w', ppser_serializer)
-    ! ELSE
-    !   CALL fs_create_serializer(directory, TRIM(prefix), 'a', ppser_serializer)
-    ! ENDIF
+    IF ( present(mpi_rank) ) THEN
+      IF (mpi_rank < 10) THEN
```

Comment:
What happens for `mpi_rank >= 100`? Isn't there a simpler way to handle all numers? We just want to join a string and a number. If there are trailing spaces we don't care because we will trim later.
It's not a big problem anyway.